### ISSUE TITLE
Json decode issues when message body is to big and spit up in multiple messages

### ISSUE THREAD

**Issue by topjor** [derglaus/snapcast-websockets-ui#1]
I have noticed on that when the message body of a json rpc message is to big, it is split in to multiple chunks.
so when this happens the interface will not load because that is currently not handled, as it is mostly only a problem when first connecting and getting the server info.

(my current setup is with 3 streams, 3 groups and 3 clients (all in seperate group currently),
 before i had this issue sometimes, sometimes not with 2 streams, 2 groups and 3 clients
 i think i'm using python 2.7, and websockify is running on a different host then client and server)

I just found this out when building my own fork of the ui
I will probably be committing my own fix shortly.

(https://github.com/topjor/snapcast-websockets-ui)

(edited: added snapcast server details)

**Comment by topjor** [comment]
My fix for my rewritten ui:
https://github.com/topjor/snapcast-websockets-ui/commit/4ab9604f210b3463da1c99033ebf6b09b8ac779a

if it can't decode the json and trows an error, it will put the message into a array, and if there are multiple chunks in the array it will merge them into 1 string and try again decoding it.
the function returns null on failing and the json decoded object if success.

**Comment by derglaus** [comment]
Hej topjor, 

I will have a look to your solution and implement it in my branch as well. Thank you very much,  I think I would have taken endless if I would have encountered the problem one day...

I have seen you based your code on my "old" version.  I updated my interface about a week ago to improve the optics a bit, but mainly to improve the usability.  I added a second slider to adjust the volume in smaller steps (see the updated Screenshot)
Only the show() and setVolume() functions should be affected as well as the css...

Have a great day!

**Comment by topjor** [comment]
thinking about it, It might be that because snapcast and the websockify server run on the same host, it skips parts of the network stack because it stays all locally (eg. the part that splits messages up in chunks), and websockets probably have a mechanism for putting messages together before handing it to the javascript part.

Also I would advise you to split things like js and css and use a consistent coding style, especially consistent indenting, as that improves readability and maintainability.

I had not thought yet about fine volume control and have not yet decided if I will add that yet, but I will keep it in mind now.
(also I just added a screenshot of my version)

**Comment by derglaus** [comment]
Yes, I agree for indenting and should also add some comments. (worked on the interface quick&dirty while moving flats and changing jobs. Plus I'm no real programmer ;)
That's also why I wanted everything in one file,  didn't use any framework, any pictures in the ui etc... Like that I didn't have to set up a Webserver but just copied the file to each phone, tablet and raspi/touchscreen. 
Still space for improvement ;)
